table/block.cc

```diff
@@ -462,7 +462,9 @@ void IndexBlockIter::DecodeCurrentValue(uint32_t shared) {
   if (shared == 0) {
     uint64_t o, s;
     const char* newp = GetVarint64Ptr(value_.data(), limit, &o);
+    assert(newp);
     newp = GetVarint64Ptr(newp, limit, &s);
+    assert(newp);
```

Comment:
Are o and s guaranteed to be initialized to valid values? Could there be some on disk corruption?

Reply:
It could. But my approach has been not to pay the cost of verifying for corruption on the fast path. This would simplify the code and also reduces branch prediction misses. The corruption should be detected by block checksums, which would be much less expensive.

Reply:
Check for nullptr means adding yet another branch. The question is that what we would gain by checking for this so frequency except paying a bit of CPU overhead? A corruption will already be caught by checksum, and if not, by a later part in the code (wrong handle pointer for example). There is not much benefit of detecting a corruption right after reading the first corrupted bit.
That being said, there is a very very very low probability of a corruption that bypasses checksums and simply report a wrong number rather than crashing because of the wrong pointer. My position is that we can live with that.